### PR TITLE
Removes instant stuns from defibs, adds knockdown and stamina damage

### PR DESCRIPTION
**Pull request by Inadvizable**
## What Does This PR Do

Emagged handheld defibs now deal 60 stamina damage and knock down for 10 seconds.
Emagged handheld defibs now have a global cooldown shared with stunbatons etc. It is 3.5 seconds.
Paddle defibs now deal 60 stamina damage and knock down for 10 seconds. 
Emagged defibs no longer have a 10% RNG chance to cause near-instant death.
Nukie combat defibs still instakill.
Non-combat uses of any of this are not affected.

I feel like handheld defibs could be done better but I suck, so if you have any suggestions on how to streamline this code, hit me with them.

## Why It's Good For The Game

Handheld defibs currently instantly sleep the target for 6 seconds. This basically kills them, especially since another defib can be chained with it. It is worse than an instant stun, since the victim cannot even call for help. This is even inconsistent with other defibs, which do not cause sleep - somehow, the weakest defib is the strongest in combat. This PR makes them strong, essentially front-loaded stunbatons (instant knockdown instead of delay), but at least not insta-lose. Giving them a cooldown prevents cheesing with multiple defibs and/or batons/prods.

Paddle defibs are an instant stun, which is bad for stamina combat, and has to be removed. However, their unwieldiness means they probably should be somewhat better than the handheld. Not having a global cooldown with batons at least gives it some sort of niche. This can be altered later if it ends up being too strong... But I think it's too unwieldy, especially without gamer macros.

The 10% chance to instakill is funny but it also sucks for fights to be decided by a 1 in 10 chance to instakill.

## Changelog
:cl:
tweak: Emagged defibs have been rebalanced to knock down and deal stamina damage. Check the PR for numbers.
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/handheld_defib.dm [hunk 1]
```diff
@@ -32,7 +38,30 @@
...
-	if(emagged || (H.health <= HEALTH_THRESHOLD_CRIT) || (H.undergoing_cardiac_arrest()))
+	if(emagged)
+		if(attack_cooldown > world.time)
+			return
+
+		var/user_UID = user.UID()
+		if(HAS_TRAIT_FROM(H, TRAIT_WAS_BATONNED, user_UID)) // no following up with baton or dual wielding defibs for stunlock cheese purposes
+			return
+
+		cooldown = world.time + (cooldown)
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/handheld_defib.dm:49]:
```suggestion
		cooldown = world.time + attack_cooldown
```
I think you meant to do this? Cooldown is a bool

Comment (Inadvizable) [reply code/game/objects/items/devices/handheld_defib.dm:49]: oh whoops, this is a leftover from something I was doing earlier that I missed - I will fix that

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/handheld_defib.dm [hunk 2]
```diff
@@ -57,18 +82,17 @@
...
 		cooldown = TRUE
 		icon_state = "[icon_base]-shock"
-		addtimer(CALLBACK(src, .proc/short_charge), 10)
+		addtimer(CALLBACK(src, .proc/short_charge), 1 SECONDS)
 		addtimer(CALLBACK(src, .proc/recharge), charge_time)
 
 	else
 		to_chat(user, "<span class='notice'>[src]'s on board medical scanner indicates that no shock is required.</span>")
 
+/obj/item/handheld_defibrillator/proc/allowhit(mob/living/target, user_UID, knockdown_duration)
```
[the remark this answers is not in the document]
Comment (Inadvizable) [reply code/game/objects/items/devices/handheld_defib.dm:93]: oh yeah oops. from when i was thinking of making these have a delay and paddles not.

File: code/game/objects/items/devices/handheld_defib.dm [hunk 3]
```diff
@@ -32,7 +38,26 @@
...
+		if(HAS_TRAIT_FROM(H, TRAIT_WAS_BATONNED, user_UID)) // no following up with baton or dual wielding defibs for stunlock cheese purposes
+			return
+
+		user.visible_message("<span class='danger'>[user] violently shocks [H] with [src]!</span>", "<span class='danger'>You violently shock [H] with [src]!</span>")
+		add_attack_logs(user, H, "emag-defibbed with [src]")
+		playsound(user.loc, "sound/weapons/Egloves.ogg", 75, 1)
+		H.KnockDown(knockdown_duration)
+		H.adjustStaminaLoss(60)
+		SEND_SIGNAL(H, COMSIG_LIVING_MINOR_SHOCK, 100)
+		ADD_TRAIT(H, TRAIT_WAS_BATONNED, user.UID())
```
[the remark this answers is not in the document]
Comment (Inadvizable) [reply code/game/objects/items/devices/handheld_defib.dm]: yeppers. and this is actually a slip up I copied from the new batons so I fixos that now

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**